src/libs/PersonalDetailsUtils.ts

```diff
@@ -24,9 +24,14 @@ Onyx.connect({
     },
 });
 
+const hiddenText = Localize.translateLocal('common.hidden');
+const substringStartIndex = 8;
 function getDisplayNameOrDefault(passedPersonalDetails?: Partial<PersonalDetails> | null, defaultValue = '', shouldFallbackToHidden = true): string {
-    const displayName = passedPersonalDetails?.displayName ? passedPersonalDetails.displayName.replace(CONST.REGEX.MERGED_ACCOUNT_PREFIX, '') : '';
-    const fallbackValue = shouldFallbackToHidden ? Localize.translateLocal('common.hidden') : '';
+    let displayName = passedPersonalDetails?.displayName ?? '';
+    if (displayName.startsWith(CONST.MERGED_ACCOUNT_PREFIX)) {
+        displayName = displayName.substring(substringStartIndex);
+    }
+    const fallbackValue = shouldFallbackToHidden ? hiddenText : '';
```

Comment:
this function is getting called a lot of times, maybe since we default to the `displayName` being returned (I assume that's the hot path) we can check whether to return it first and only then declare the additional `fallbackValue` constant we fallback to? Might save us some memory being allocated.

Reply:
I agree and nice spotting 👍